### ISSUE TITLE
Shipkit for Scala multi-variant project

### ISSUE THREAD

**Issue by mockitoguy** [mockito/shipkit#870]
This ticket is prescribe a clean solution for a Scala multi-variant project such as https://github.com/linkedin/isolation-forest. Currently, this project uses Shipkit but the setup is complex (multiple Travis CI stages, disabling various Shipkit tasks - https://github.com/linkedin/isolation-forest/blob/master/.travis.yml#L21). More context: #858 

isolation-forest requirements, must-haves:
 - automated release to Bintray

nice-to-haves:
 - release notes
 - avoiding unnecessary release (e.g. "releaseNeeded")

**Comment by mockitoguy** [comment]
@jverbus, can you review this ticket and confirm the requirements? (I'll update the description).

**Comment by mockitoguy** [comment]
@jverbus, any thoughts? We could simplify isolation-forest automation by using those 2 plugins:
 - https://github.com/shipkit/shipkit-auto-version
 - https://github.com/shipkit/shipkit-changelog

**Comment by jverbus** [comment]
Thanks for creating the new ticket!

Will this require a cross-build setup so that all variants are built simultaneously, or will it be possible to have multiple separate build jobs in the same Travis stage with different build parameters that all get released with the same version?

**Comment by mockitoguy** [comment]
> Will this require a cross-build setup so that all variants are built simultaneously

No. The new plugins are simple and (mostly) unopinionated. It should be fairly easy to set up your Travis CI automation based on *your* requirements, and have the plugins help you with versioning and release notes generation.

Go ahead and try it, and tell us how is it going.
